src/maggma/stores/open_data.py

```diff
@@ -1,19 +1,324 @@
+
+        Raises:
+            AttributeError: if the key and last updated fields are not both present in this store or
+                if criteria is provided when exhaustive is not set to True
+        """
+        if not (self._field_exists(self.key) and self._field_exists(self.last_updated_field)):
+            raise AttributeError("This index store does not contain data with both key and last updated fields")
+
+        if criteria is not None and not exhaustive:
+            raise AttributeError("Criteria is only considered when doing an item-by-item check")
+
+        if exhaustive:
+            # Get our current last_updated dates for each key value
+            props = {self.key: 1, self.last_updated_field: 1, "_id": 0}
+            dates = {
+                d[self.key]: self._lu_func[0](d.get(self.last_updated_field, datetime.max))
+                for d in self.query(properties=props)
+            }
+
+            # Get the last_updated for the store we're comparing with
+            props = {target.key: 1, target.last_updated_field: 1, "_id": 0}
+            target_dates = {
+                d[target.key]: target._lu_func[0](d.get(target.last_updated_field, datetime.min))
+                for d in target.query(criteria=criteria, properties=props)
+            }
+
+            new_keys = set(target_dates.keys()) - set(dates.keys())
+            updated_keys = {key for key, date in dates.items() if target_dates.get(key, datetime.min) > date}
+
+            return list(new_keys | updated_keys)
+
+        criteria = {"query": f"{self.last_updated_field} > '{self._lu_func[1](self.last_updated)}'"}
+        return target.distinct(field=self.key, criteria=criteria)
+
+    def distinct(self, field: str, criteria: Optional[Dict] = None, all_exist: bool = False) -> List:
+        """
+        Get all distinct values for a field
+
+        Args:
+            field: the field(s) to get distinct values for
+            criteria: the value of the `query` key will be used as the Pandas string expression to query with;
+                all other data will be ignored
+
+        Returns:
+            List: a list of all the distinct values for the provided field (after filtering by the provided criteria)
+        """
+        criteria = criteria or {}
+
+        return [key for key, _ in self.groupby(field, properties=[field], criteria=criteria)]
+
+    def groupby(
+        self,
+        keys: Union[List[str], str],
+        criteria: Optional[Dict] = None,
+        properties: Union[Dict, List, None] = None,
+        sort: Optional[Dict[str, Union[Sort, int]]] = None,
+        skip: int = 0,
+        limit: int = 0,
+    ) -> Iterator[Tuple[Dict, List[Dict]]]:
+        """
+        Simple grouping function that will group documents
+        by keys.
+
+        Args:
+            keys: fields to group documents
+            criteria: PyMongo filter for documents to search in
```

Comment:
`criteria` here shouldn't be a PyMongo filter, right? Should be same doc string as above?
`criteria: the value of the `query` key will be used as the Pandas string expression to query with;`

Reply:
fix added - thanks for catching this